client/coral-embed-stream/src/CommentStream.js

```diff
@@ -284,20 +279,14 @@ const mapStateToProps = state => ({
   userData: state.user.toJS()
 });
 
-const mapDispatchToProps = (dispatch) => ({
-  addItem: (item, itemType) => dispatch(addItem(item, itemType)),
-  updateItem: (id, property, value, itemType) => dispatch(updateItem(id, property, value, itemType)),
-  postItem: (data, type, id) => dispatch(postItem(data, type, id)),
-  getStream: (rootId) => dispatch(getStream(rootId)),
-  addNotification: (type, text) => dispatch(addNotification(type, text)),
-  clearNotification: () => dispatch(clearNotification()),
-  showSignInDialog: (offset) => dispatch(showSignInDialog(offset)),
-  postAction: (item, action, user, itemType) => dispatch(postAction(item, action, user, itemType)),
-  deleteAction: (item, action, user, itemType) => dispatch(deleteAction(item, action, user, itemType)),
-  appendItemArray: (item, property, value, addToFront, itemType) => dispatch(appendItemArray(item, property, value, addToFront, itemType)),
-  handleSignInDialog: () => dispatch(authActions.showSignInDialog()),
-  logout: () => dispatch(logout()),
-  updateStatus: status => dispatch(updateOpenStatus(status))
-});
+// Ties all imported actions to dispatch and makes them available in props.
+const mapDispatchToProps = (dispatch) => {
+  const actions = Object.assign({}, itemActions, notificationActions, authActions, configActions);
+
+  return Object.keys(actions).reduce((dispatchActions, key) => {
```

Comment:
That seems fine, I'm interested in:

1) Defining and sticking to some established API of actions that we make available to plugin developers eventually.
2) Not asking plugin developers to worry about a mapDispatchToProps or mapStateToProps function. 

It may be that there are multiple parts of our application (in-stream admin settings, the main comment stream, user settings, etc) that a plugin developer can connect into. If so we should maybe draw a distinction between a part of the application (`coral-sign-in`) that we don't really intend to be replaceable by a plugin and a part of the architecture (`coral-plugin-like-button`) that we do.